icu4c/source/common/locid.cpp

```diff
@@ -1422,20 +1422,22 @@ AliasReplacer::replaceTerritory(UVector& toBeFreed, UErrorCode& status)
                          *(foundInReplacement-1) == ' ');
                 U_ASSERT(foundInReplacement[len] == ' ' ||
                          foundInReplacement[len] == '\0');
-                item = new CharString(foundInReplacement, (int32_t)len, status);
+                item.adoptInsteadAndCheckErrorCode(
+                    new CharString(foundInReplacement, (int32_t)len, status), status);
             }
         }
-        if (item == nullptr) {
-            item = new CharString(replacement,
-                                  (int32_t)(firstSpace - replacement), status);
+        if (item.isNull()) {
```

Comment:
```suggestion
        if (item.isNull() && U_SUCCESS(status)) {
```
We should not try to create a new CharString if something went wrong.

I suggest using `git commit --amend` to keep it to one commit.

Reply:
DONE

Reply:
PTAL

Reply:
Sorry, looking at GitHub I don't see this line changing. What am I missing?

Reply:
git commit --amend didn't work as I expected.